Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
```

Comment:
> The exe is written as a .Net Framework application (not sure how well it actually tests CoreFX)

What do you mean by this?  How could it be a .NET Framework app if it runs on Linux and macOS, as it does?

> Exceptions are not properly marshalled back and when the app crashes we get no relevant information.

Sure we do... it's in the output log, it's just not as convenient as it could be.

> This currently blocks UWP testing (internal thread since we're still working on the tooling) and there is no foreseeable fix for the problem.

Then that's a UWP-testing issue we need to figure out how to solve.  There are very real and genuine reasons to run things out of process.